llama_index/query_engine/jsonalyze_query_engine.py

```diff
@@ -0,0 +1,236 @@
+        self._response_synthesis_prompt = (
+            response_synthesis_prompt or DEFAULT_RESPONSE_SYNTHESIS_PROMPT
+        )
+        self._table_name = table_name
+        self._verbose = verbose
+
+        super().__init__(self._service_context.callback_manager)
+
+    def _get_prompts(self) -> Dict[str, Any]:
+        """Get prompts."""
+        return {
+            "jsonalyze_prompt": self._jsonalyze_prompt,
+            "response_synthesis_prompt": self._response_synthesis_prompt,
+        }
+
+    def _update_prompts(self, prompts: PromptDictType) -> None:
+        """Update prompts."""
+        if "jsonalyze_prompt" in prompts:
+            self._jsonalyze_prompt = prompts["jsonalyze_prompt"]
+        if "response_synthesis_prompt" in prompts:
+            self._response_synthesis_prompt = prompts["response_synthesis_prompt"]
+
+    def _get_prompt_modules(self) -> PromptMixinType:
+        """Get prompt sub-modules."""
+        return {}
+
+    def _query(self, query_bundle: QueryBundle) -> Response:
+        """Answer an analytical query on the JSON List."""
+        query = query_bundle.query_str
+        if self._verbose:
+            print_text(f"Query: {query}\n", color="green")
+
+        # Perform the analysis
+        sql_query, table_schema, results = self._analyzer(
+            self._list_of_dict,
+            query,
+            self._service_context,
+            table_name=self._table_name,
+            prompt=self._jsonalyze_prompt,
+        )
+        if self._verbose:
+            print_text(f"SQL Query: {sql_query}\n", color="blue")
+            print_text(f"Table Schema: {table_schema}\n", color="cyan")
+            print_text(f"SQL Response: {results}\n", color="yellow")
+
+        if self._synthesize_response:
+            response_str = self._service_context.llm.predict(
+                self._response_synthesis_prompt,
+                sql_query=sql_query,
+                table_schema=table_schema,
+                sql_response=results,
+                query_str=query_bundle.query_str,
+            )
+            if self._verbose:
+                print_text(f"Response: {response_str}", color="magenta")
+        else:
+            response_str = json.dumps(
+                {
+                    "sql_query": sql_query,
+                    "table_schema": table_schema,
+                    "sql_response": results,
+                }
+            )
+        response_metadata = {"sql_query": sql_query, "table_schema": str(table_schema)}
+
+        return Response(response=response_str, metadata=response_metadata)
+
+    async def _aquery(self, query_bundle: QueryBundle) -> Response:
+        """Answer an analytical query on the JSON List."""
+        query = query_bundle.query_str
+        if self._verbose:
+            print_text(f"Query: {query}", color="green")
+
+        # Perform the analysis
+        sql_query, table_schema, results = self._analyzer(
```

Comment:
Should we have an async analyzer function?

Reply:
Sorry for missing it, done:)